pass.go

```diff
@@ -0,0 +1,834 @@
+	PageTitle         string
+	CSSIntegrityHash  string
+	JSIntegrityHash   string
+	TemplateVariables interface{}
+}
+
+// TemplateVariables TODO split to template specific structs
+type TemplateVariables struct {
+	FormURL     string `json:"-"`
+	URL         string `json:"url,omitempty"`
+	UUID        string `json:"uuid,omitempty"`
+	Recipient   string `json:"recipient,omitempty"`
+	SenderID    string `json:"senderid,omitempty"`
+	Secret      string `json:"secret,omitempty"`
+	ErrorString string `json:"-"`
+}
+
+var contentSecurityPolicyHTML = strings.Join([]string{
+	"default-src 'none';",
+	"img-src 'self';",
+	"style-src 'self';",
+	"script-src 'self';",
+	"connect-src 'self';",
+	"form-action 'self';",
+	"frame-ancestors 'none';",
+	"block-all-mixed-content;",
+	"sandbox allow-scripts allow-forms allow-same-origin;",
+	"require-sri-for script style;",
+	"base-uri 'none';",
+}, " ")
+
+var contentSecurityPolicyJSON = strings.Join([]string{
+	"default-src 'none';",
+	"frame-ancestors 'none';",
+	"block-all-mixed-content;",
+	"sandbox",
+	"require-sri-for script style;",
+	"base-uri 'none';",
+}, " ")
+
+var featurePolicy = strings.Join([]string{
+	"accelerometer 'none';",
+	"ambient-light-sensor 'none';",
+	"autoplay 'none';",
+	"camera 'none';",
+	"encrypted-media 'none';",
+	"fullscreen 'none';",
+	"geolocation 'none';",
+	"gyroscope 'none';",
+	"magnetometer 'none';",
+	"microphone 'none';",
+	"midi 'none';",
+	"payment 'none';",
+	"picture-in-picture 'none';",
+	"speaker 'none';",
+	"sync-xhr 'none';",
+	"sync-script 'none';",
+	"usb 'none';",
+	"vr 'none';",
+}, " ")
+
+// generateSecurityHeaders generates the http securityHeaders to be outputted
+// as http headers before every request
+func (p *Pass) generateSecurityHeaders() {
+	p.sv.securityHeaders = map[string]string{}
+	// HSTS
+	p.sv.securityHeaders["Strict-Transport-Security"] = fmt.Sprintf("max-age=%d", p.sv.cfg.HSTS.MaxAge)
+	if p.sv.cfg.HSTS.IncludeSubDomains {
+		p.sv.securityHeaders["Strict-Transport-Security"] += "; includeSubDomains"
+	}
+	if p.sv.cfg.HSTS.Preload {
+		p.sv.securityHeaders["Strict-Transport-Security"] += "; preload"
+	}
+
+	p.sv.securityHeaders["Access-Control-Allow-Origin"] = p.GetURL()
+	p.sv.securityHeaders["Feature-Policy"] = featurePolicy
+	p.sv.securityHeaders["Referrer-Policy"] = "no-referrer"
+	p.sv.securityHeaders["X-Content-Type-Options"] = "nosniff"
+	p.sv.securityHeaders["X-Frame-Options"] = "deny"
+	p.sv.securityHeaders["X-XSS-Protection"] = "1; mode=block"
+}
+
+// serverError takes in any error this program is designed to handle.
+//
+// It returns if the error is a server side problem, or something the user sent
+// that was malformed, and the string to be displayed to the user, unless we
+// choose not to display it for obscurity.
+func (p *Pass) serverError(err error) (statusCode int, t *template.Template, txt string) {
+	switch e := err.(type) {
+	case net.Error:
+		if opErr, okOpErr := e.(*net.OpError); okOpErr {
```

Comment:
Fold this `if` into the outer `switch ... case`.

Reply:
Fixed in next push